packages/main/src/SegmentedButtonItem.ts

```diff
@@ -164,6 +171,10 @@ class SegmentedButtonItem extends UI5Element implements IButton, ISegmentedButto
 		}
 	}
 
+	onBeforeRendering(): void {
+		this.iconOnly = this.isIconOnly;
```

Comment:
IMO it would be best to replace the `isIconOnly` getter entirely with the new `iconOnly` property, because they are the same thing after this change. So:

`this.iconOnly = !willShowContent(this.text)`

in `onBeforeRendering`, delete the getter, and use `iconOnly` directly whenever the getter was used.